### PR TITLE
feat: should emit stream error event when file too large

### PR DESCRIPTION
**Pull request by fengmk2**
closes https://github.com/eggjs/egg/issues/590

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: app/extend/context.js [hunk 1]
```diff
@@ -42,6 +42,16 @@ module.exports = {
...
       stream.fields = parts.field;
+      stream.once('limit', () => {
+        const err = new Error('Request file too large');
+        err.name = 'MultipartFileTooLargeError';
+        err.status = 413;
+        err.fields = stream.fields;
+        stream.emit('error', err);
+        stream.unpipe();
+        stream.removeAllListeners('data');
+        stream.resume();
```
Comment (dead-horse) [inline app/extend/context.js]: resume 可以让客户端终止请求么？

Comment (fengmk2) [inline app/extend/context.js]: 不会，就是相当于黑洞

File: app/extend/context.js [hunk 2]
```diff
@@ -42,6 +42,16 @@ module.exports = {
         ctx.throw(400, 'Can\'t found upload file');
       }
       stream.fields = parts.field;
+      stream.once('limit', () => {
+        const err = new Error('Request file too large');
+        err.name = 'MultipartFileTooLargeError';
+        err.status = 413;
+        err.fields = stream.fields;
+        stream.emit('error', err);
```
Comment (fengmk2) [inline app/extend/context.js]: 打个日志，有lisener 才触发 error

**Comment by popomore** [comment]
覆盖率

**Comment by fengmk2** [comment]
1.2.0

**Pull request merged**